web/views.py

```
@@ -6654,6 +6679,65 @@
+        try:
+            send_mail(
+                subject,
+                message,
+                settings.DEFAULT_FROM_EMAIL,
+                [session.course.teacher.email],
+                fail_silently=True,
+            )
+        except Exception:
+            logger.exception("Failed to send attendance notification email")
+    else:
+        logger.warning(f"No email found for teacher of session {session.id}, skipping attendance notification")
```

Ruff (0.8.2)

Logging statement uses f-string

(G004)